auth/standard_role_manager.cc

```diff
@@ -340,87 +334,79 @@ standard_role_manager::alter(std::string_view role_name, const role_config_updat
 }
 
 future<> standard_role_manager::drop(std::string_view role_name) {
-    return this->exists(role_name).then([this, role_name](bool role_exists) {
-        if (!role_exists) {
-            throw nonexistant_role(role_name);
+    if (!co_await this->exists(role_name)) {
+        throw nonexistant_role(role_name);
+    }
+    // First, revoke this role from all roles that are members of it.
+    const auto revoke_from_members = [this, role_name] () -> future<> {
+        static const sstring query = format("SELECT member FROM {}.{} WHERE role = ?",
+                get_auth_ks_name(_qp),
+                meta::role_members_table::name);
+        const auto members = co_await _qp.execute_internal(
+                query,
+                consistency_for_role(role_name),
+                internal_distributed_query_state(),
+                {sstring(role_name)},
+                cql3::query_processor::cache_internal::no);
+        co_await parallel_for_each(
+                members->begin(),
+                members->end(),
+                [this, role_name] (const cql3::untyped_result_set_row& member_row) -> future<> {
+                    const sstring member = member_row.template get_as<sstring>("member");
+                    co_await this->modify_membership(member, role_name, membership_change::remove);
+                }
+        );
+    };
+    // In parallel, revoke all roles that this role is members of.
+    const auto revoke_members_of = [this, grantee = role_name] {
+        return this->query_granted(
+                grantee,
+                recursive_role_query::no).then([this, grantee](role_set granted_roles) {
+            return do_with(
```

Comment:
Since we're coroutinizing, why not coroutinize `revoke_members_of` too? We could get rid of the `do_with`.

Reply:
ok